sqlglot/executor/python.py

```diff
@@ -286,23 +286,29 @@ def aggregate(self, step, context):
     def sort(self, step, context):
         projections = self.generate_tuple(step.projections)
 
-        sink = self.table(step.projections)
+        projection_columns = [p.alias_or_name for p in step.projections]
+        all_columns = list(context.columns) + projection_columns
+        sink = self.table(all_columns)
```

Comment:
what if they columns have the same name? do you need to use column_range?

Reply:
Projections get the `None` table in `sort_ctx`. Everything else is nested by the table name. (You wrote that logic haha)

And the optimizer handles qualifying columns for us. So it just works :)



I added a test to demonstrate what happens.

```sql
SELECT 
  100 - ORD(a) AS a, 
  a AS i 
FROM x 
ORDER BY a
```

returns:
```python
[(1, "c"), (2, "b"), (3, "a")]
```